### DISCUSSION TITLE
Modules and State

### DISCUSSION THREAD

**Discussion by ericlaaper** [permafrost-dev/vue-ray#16] [Q&A] [answered]
Hi there I really like ray with laravel and want to use it also for Vue and specially Vuex. I am using multiple stores as modules. In Ray, I can track the mutations but not the state. In ray Vuex state in an empty object. 
Help is highly appreciated.

Cheers,

Eric

**This my vuex file:**

import Vue from "vue";
import Vuex from "vuex";
import { RayVuexPlugin } from "vue-ray";
import * as auth from "@/store/modules/Auth";
import * as user from "@/store/modules/User/User";
import * as message from "@/store/modules/Message";
import * as usersingle from "@/store/modules/User/UserSingle";
import * as team from "@/store/modules/Team/Team";
Vue.use(Vuex);

export default new Vuex.Store({
  strict: true,
  modules: {
    auth,
    user,
    message,
    usersingle,
    team,
},
  plugins: [
    RayVuexPlugin({
      trackState: true,
      logMutations: true,
      trackingOptions: {
        propNames: ["on*"],
      },
    }),
  ],
});

**One off the modules**
import { getError } from "@/utils/helpers";
import TeamService from "@/services/TeamService";
export const namespaced = true;

export const state = {
  teams: [],
  loading: false,
  error: null,
  status: null,
};

export const mutations = {
  SET_TEAMS(state, teams) {
    state.teams = teams;
  },
  SET_LOADING(state, loading) {
    state.loading = loading;
  },
  SET_ERROR(state, error) {
    state.error = error;
  },
};

export const actions = {
  getTeams({ commit }) {
    commit("SET_LOADING", true);

    TeamService.getTeams()
      .then((response) => {
        commit("SET_TEAMS", response.data.data);
      })
      .catch((error) => {
        commit("SET_LOADING", false);
        commit("SET_ERROR", getError(error));
      })
      .finally(() => {
        commit("SET_LOADING", false);
      });
  },
  setDefault({ commit }) {
    commit("SET_LOADING", false);
    commit("SET_ERROR", null);
    commit("SET_TEAMS", []);
  },
};
export const getters = {
  teams: (state) => {
    return state.teams;
  },
  loading: (state) => {
    return state.loading;
  },
  error: (state) => {
    return state.error;
  },

};

**Comment by patinthehat** [comment, marked as answer]
Currently tracking Vuex with multiple stores isn't supported, however it will be added to a future release.
